sceptre/plan/actions.py

```diff
@@ -1000,3 +1001,13 @@ def diff(self, stack_differ):
         :rtype: sceptre.diffing.stack_differ.StackDiff
         """
         return stack_differ.diff(self)
+
+    @add_stack_hooks
+    def stack_name(self):
+        """
+        Returns the Stack's stack name.
+
+        :returns: The Stack's stack name (external_name).
+        :rtype: str
+        """
+        return f"{self.stack.external_name} (from: {self.rel_path})"
```

Comment:
The "from" bit seems unnecessary. This appears to be a _front-end_, presentation concern that you've injected into the StackActions. Since the SceptrePlan ends up making a dictionary of Stacks with the action return value as the value anyway, I think that if you really wanted this output, that would be put together by the CLI function, not in the Actions.

Reply:
Honestly, I don't think this needs to be an Action at all.